pkg/lib/cockpit-components-table.tsx

```diff
@@ -65,11 +69,60 @@ const _ = cockpit.gettext;
  *   event, isSelected, rowIndex, rowData, extraData. rowData contains props with an id property of the clicked row.
  * - onHeaderSelect: event, isSelected.
  */
+
+interface ListingTableRowColumnProps {
+    title: React.ReactNode;
+    sortKey?: string;
+    props?: TdProps | ThProps;
+}
+
+type ListingTableRowColumn = React.ReactNode | ListingTableRowColumnProps;
+
+function cell_is_props(cell: ListingTableRowColumn): cell is ListingTableRowColumnProps {
+    return typeof cell == "object" && (cell as ListingTableRowColumnProps).title !== undefined;
+}
+
+export interface ListingTableRowProps {
+    columns: ListingTableRowColumn[];
+    props?: TrProps;
+    expandedContent?: React.ReactNode;
+    selected?: boolean;
+    initiallyExpanded?: boolean;
+    hasPadding?: boolean;
+}
+
+export interface ListingTableColumnProps {
+    title: string;
+    header?: boolean;
+    sortable?: boolean;
+    props?: ThProps;
+}
+
+export interface ListingTableProps extends Omit<TableProps, 'rows' | 'onSelect'> {
+    actions?: React.ReactNode[],
+    afterToggle?: (expanded: boolean) => void,
+    caption?: string,
+    className?: string,
+    columns: (string | ListingTableColumnProps)[],
+    emptyCaption?: string,
+    emptyCaptionDetail?: React.ReactNode,
```

Comment:
Seems to be a string as well? Or better said only a string:

```
                        <TextContent>
                            <Text component={TextVariants.small}>
                                {emptyCaptionDetail}
                            </Text>
                        </TextContent>
```

Reply:
Only used here https://github.com/cockpit-project/cockpit-machines/blob/0a413a926458b7a3bdfe1a218ce3b41790a20801/src/components/vm/snapshots/vmSnapshotsCard.jsx#L245

Reply:
Also showing useless html when its empty ;-)

Reply:
> Seems to be a string as well?

Right, or both `emptyCaption` and `emptyCaptionDetail` are ReactNodes... Let's make them nodes, ListingTable is fine with that.

Reply:
I don't see that, emptyCaptionDetail is a string imo

Reply:
ListingTable only uses it as the "children" property of Text, which is ReactNode. I guess it is legal have `<Text><b>Hi</b><Text>`...

Reply:
Right it converts it to `<small>` and I guess you can put a <b> in there but then it would be ReactNode | string no?